pallets/gear/src/schedule.rs

```diff
@@ -660,107 +658,72 @@ impl<T: Config> From<MemoryWeights<T>> for LazyPagesCosts {
     }
 }
 
-macro_rules! replace_token {
-    ($_in:tt $replacement:tt) => {
-        $replacement
-    };
-}
-
-macro_rules! call_zero {
-    ($name:ident, $( $arg:expr ),*) => {
-        <T as Config>::WeightInfo::$name($( replace_token!($arg 0) ),*)
-    };
-}
-
-macro_rules! cost_args {
-    ($name:ident, $( $arg: expr ),+) => {
-        (<T as Config>::WeightInfo::$name($( $arg ),+).saturating_sub(call_zero!($name, $( $arg ),+))).ref_time()
-    }
-}
-
-macro_rules! cost_batched_args {
-    ($name:ident, $( $arg: expr ),+) => {
-        cost_args!($name, $( $arg ),+) / u64::from(API_BENCHMARK_BATCH_SIZE)
-    }
-}
-
-macro_rules! cost_instr_no_params_with_batch_size {
-    ($name:ident, $batch_size:expr) => {
-        (cost_args!($name, 1) / u64::from($batch_size)) as u32
-    };
-}
-
-macro_rules! cost_instr_with_batch_size {
-    ($name:ident, $num_params:expr, $batch_size:expr) => {
-        cost_instr_no_params_with_batch_size!($name, $batch_size).saturating_sub(
-            (cost_instr_no_params_with_batch_size!(instr_i64const, $batch_size) / 2)
-                .saturating_mul($num_params),
-        )
-    };
+#[inline]
+fn cost(w: fn(u32) -> Weight) -> Weight {
```

Comment:
If there are any performance concerns about whether the compiler will be able to reduce these function calls to simple assignments when passing a pointer to a function, it could be replaced with impl Fn(u32) -> Weight. However, this doesn't seem to be the case.